evap/student/views.py

```diff
@@ -149,6 +149,7 @@ def get_valid_form_groups_or_render_vote_page(request, evaluation, preview, for_
 
 @participant_required
 def vote(request, evaluation_id):
+    # pylint: disable=too-many-locals,too-many-nested-blocks,too-many-branches
```

Comment:
Maybe we should consider opening a refactor issue?

Reply:
I looked through it and I think the code pretty much states what needs to be done, not less, not more. If anyone finds a way to make this simpler, sure go ahead! But splitting it up into multiple methods would, in my opinion, not make it simpler. It would just introduce a bunch of parameters that one also needs to read and understand, and I'd do it only if it helped with something else, e.g. if we could reasonably unit-test the individual methods.